controllers/replace_misconfigured_pods.go

```diff
@@ -227,53 +229,55 @@ func instanceNeedsRemoval(cluster *fdbtypes.FoundationDBCluster, instance FdbIns
 			return false, err
 		}
 
-		for idx, container := range desiredSpec.Containers {
-			if equality.Semantic.DeepEqual(container.Resources, instance.Pod.Spec.Containers[idx].Resources) {
-				continue
-			}
-
-			if resourcesNeedsReplacement(container.Resources, instance.Pod.Spec.Containers[idx].Resources) {
-				log.Info("Replace instance",
-					"namespace", cluster.Namespace,
-					"name", cluster.Name,
-					"processGroupID", instanceID,
-					"reason", "Resource requests have changed")
-				return true, nil
-			}
+		if resourcesNeedsReplacement(desiredSpec.Containers, instance.Pod.Spec.Containers) {
+			log.Info("Replace instance",
+				"namespace", cluster.Namespace,
+				"name", cluster.Name,
+				"processGroupID", instanceID,
+				"reason", "Resource requests have changed")
+			return true, nil
 		}
 
-		for idx, container := range desiredSpec.InitContainers {
-			if equality.Semantic.DeepEqual(container.Resources, instance.Pod.Spec.InitContainers[idx].Resources) {
-				continue
-			}
-
-			if resourcesNeedsReplacement(container.Resources, instance.Pod.Spec.InitContainers[idx].Resources) {
-				log.Info("Replace instance",
-					"namespace", cluster.Namespace,
-					"name", cluster.Name,
-					"processGroupID", instanceID,
-					"reason", "Resource requests have changed")
-				return true, nil
-			}
+		if resourcesNeedsReplacement(desiredSpec.InitContainers, instance.Pod.Spec.InitContainers) {
```

Comment:
Thinking about this again we could be even smarter and check if one of init container wants more resources than all other containers and only then trigger a replacement (.e.g if the actual required resources increase) but I think the current solution should be fine and we can revisit this one if needed.